src/plugins/embeddable/public/embeddable_panel/use_select_from_embeddable.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useEffect, useState } from 'react';
+import { distinctUntilKeyChanged } from 'rxjs';
+import { EmbeddableInput, EmbeddableOutput, IEmbeddable } from '../lib';
+
+export const useSelectFromOptionalEmbeddableInput = <
```

Comment:
I am not sure useSelectFromOptionalEmbeddableInput is the best name. Its only used once (see code below), and its not for optional key, its for optional embeddable. How about just removing `useSelectFromEmbeddableInput` and renaming `useSelectFromOptionalEmbeddableInput` to `useSelectFromEmbeddableInput`?

```
const parentHidePanelTitle = useSelectFromOptionalEmbeddableInput(
    'hidePanelTitles',
    embeddable.parent
  );
```

Reply:
That's not a bad idea - I think I anticipated using that more. One thing that removing `useSelectFromOptionalEmbeddableInput` will do is mean that any time you select from the embeddable input it'll be `type | undefined`. I'm not sure if that would cause any problems - probably will just mean some more guards. I'll implement this!